airflow/providers_manager.py

```diff
@@ -51,6 +52,9 @@
 else:
     from importlib_resources import files as resource_files
 
+PS = ParamSpec("PS")
+RT = TypeVar("RT")
```

Comment:
I'm not ok with doing these typings in this PR

Reply:
This required to fix `mypy`, because right now it doens't work correctly due to breaking by statica hack, see: https://github.com/apache/airflow/pull/39430#discussion_r1590962541

This already moved into the separate PR: https://github.com/apache/airflow/pull/39750